File: kotlin-semver/karakum/plugins/convertIdentifierBasePackage.js
import ts from "typescript";
import * as karakum from "karakum";

export default function (node, context) {
    const typeScriptService = context.lookupService(karakum.typeScriptServiceKey)

    if (
        ts.isTypeReferenceNode(node)
        && typeScriptService.printNode(node) === "inc.IdentifierBase"
    ) {
        return "IdentifierBase"
    }

    return null
}

File: kotlin-semver/karakum/src/jsMain/kotlin/semver/karakum/plugins/convertIdentifierBasePackage.kt
package semver.karakum.plugins

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isIdentifier
import typescript.isQualifiedName
import typescript.isTypeReferenceNode

val convertIdentifierBasePackage = createPlugin { node, _, _ ->
    nullable {
        ensure(isTypeReferenceNode(node))

        val typeName = node.typeName
        ensure(isQualifiedName(typeName))

        val left = typeName.left
        ensure(isIdentifier(left))
        ensure(left.text == "inc")

        ensure(typeName.right.text == "IdentifierBase")

        "IdentifierBase"
    }
}
